pkg/provider/azure_loadbalancer.go

```diff
@@ -227,9 +228,12 @@ func (az *Cloud) cleanupVMSetFromBackendPoolByCondition(slb *network.LoadBalance
 					ipConf := (*bp.BackendIPConfigurations)[i]
 					ipConfigID := to.String(ipConf.ID)
 					_, vmSetName, err := az.VMSet.GetNodeNameByIPConfigurationID(ipConfigID)
-					if err != nil {
+					if err != nil && !errors.Is(err, cloudprovider.InstanceNotFound) {
 						return nil, err
 					}
+					if vmSetName == "" {
+						continue
```

Comment:
If it is `cloudprovider.InstanceNotFound` that make the `vmSetName` empty, I think we continue to remove the IP config from the backend pool?